### DISCUSSION TITLE
[firebase_storage] add support for sessionUri

### DISCUSSION THREAD

**Discussion by wybie-lovat** [firebase/flutterfire#5564] [Feature request]
**Is your feature request related to a problem? Please describe.**
When the app is closed, there's no way to resume an upload task using the sessionUri same as in the native implementations, so I have to restart the upload

**Describe the solution you'd like**
Adding `sessionUri` as a param to `putFile`

**Additional context**
The events snapshot already contain the `sessionUri`

**Comment by wybie-lovat** [comment]
Is this something you guys will be willing to accept a PR for?

**Comment by wybie-lovat** [comment]
Hey, any updates?

**Comment by paurakhsharma** [comment]
#724 is closed saying it is already implemented but it is not implemented. 
Looking at the native documentation I can see that `resume()` and `Continuing Uploads Across Process Restarts` using `uploadSessionUri` as two different things. 

https://firebase.google.com/docs/storage/android/upload-files#manage_uploads

https://firebase.google.com/docs/storage/android/upload-files#continuing_uploads_across_process_restarts

`resume()` only resumes a `pause()`ed uploads not the one after application restarts.

**Comment by Ehesp** [comment]
I've taken a look into this as part of the rework, it wont be landing in the initial rework but there's a potential for it to be implemented further down the line. Some observations:

- It's an Android only API.
- Restoring tasks is easy enough, however integrating this in Dart side is trickier.. If the app resumes from a state the Dart code is restarted and we'd need to recreate the tasks on the back of the resumed ones.
- "The events snapshot already contain the sessionUri", this is true, however they are currently unused.

**Comment by gusmd** [comment]
Also really interested in this feature, as currently a developer needs to maintain a lot of extra logic to support the case where the app is closed while uploading.

Curious if there has been any progress on this?

**Reply by Ehesp** [reply]
No progress - right now we need it to be a feature across Android, iOS & Web for it to be considered.
